crates/paths/src/lib.rs

```diff
@@ -45,20 +65,30 @@ impl Borrow<AbsPath> for AbsPathBuf {
     }
 }
 
+impl TryFrom<Utf8PathBuf> for AbsPathBuf {
+    type Error = Utf8PathBuf;
+    fn try_from(path_buf: Utf8PathBuf) -> Result<AbsPathBuf, Utf8PathBuf> {
+        if !path_buf.is_absolute() {
+            return Err(path_buf);
+        }
+        Ok(AbsPathBuf(path_buf))
+    }
+}
+
 impl TryFrom<PathBuf> for AbsPathBuf {
```

Comment:
It feels like this impl should now be deleted? Not it is `try` for two orthogonal reasons:

* either the path is not valid utf-8
* or the path is not absolute

Better let the caller to explictily handle both of the possiblieties and do something like

```
AbsPathBuf::try_from(Utf8PathBuf::try_from(PathBuf)?)
```